internal/services/mongodb/instance.go

```diff
@@ -0,0 +1,405 @@
+package mongodb
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	mongodb "github.com/scaleway/scaleway-sdk-go/api/mongodb/v1alpha1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+	"github.com/scaleway/terraform-provider-scaleway/v2/internal/dsf"
+	"github.com/scaleway/terraform-provider-scaleway/v2/internal/httperrors"
+	"github.com/scaleway/terraform-provider-scaleway/v2/internal/locality/regional"
+	"github.com/scaleway/terraform-provider-scaleway/v2/internal/locality/zonal"
+	"github.com/scaleway/terraform-provider-scaleway/v2/internal/services/account"
+	"github.com/scaleway/terraform-provider-scaleway/v2/internal/types"
+)
+
+func ResourceInstance() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: ResourceInstanceCreate,
+		ReadContext:   ResourceInstanceRead,
+		UpdateContext: ResourceInstanceUpdate,
+		DeleteContext: ResourceInstanceDelete,
+		Timeouts: &schema.ResourceTimeout{
+			Create:  schema.DefaultTimeout(defaultMongodbInstanceTimeout),
+			Update:  schema.DefaultTimeout(defaultMongodbInstanceTimeout),
+			Delete:  schema.DefaultTimeout(defaultMongodbInstanceTimeout),
+			Default: schema.DefaultTimeout(defaultMongodbInstanceTimeout),
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Name of the MongoDB cluster",
+			},
+			"version": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "MongoDB version of the instance",
+				ConflictsWith: []string{
+					"snapshot_id",
+				},
+			},
+			"node_number": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				Description: "Number of nodes in the instance",
+			},
+			"node_type": {
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      "Type of node to use for the instance",
+				DiffSuppressFunc: dsf.IgnoreCase,
+			},
+			"user_name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
```

Comment:
Yes, the user_name cannot be updated at the moment, so changing it will indeed trigger the recreation of the whole instance.

Reply:
doing nothing be a better solution?

Reply:
Maybe having a dedicated resource for user, it is early but I would feel more comfortable with that rather than having a forcenew that could have large consequences if handled badly